README
Dia-SQL

Parse::Dia::SQL converts Dia class diagrams into SQL. Dia is a diagram
creation program for Linux, Unix and Windows released under the GPL
license.

Parse::Dia::SQL is based on tedia2sql by Tim Ellis and others.  See the
AUTHORS file for details.

INSTALLATION

To install this module, run the following commands:

	perl Makefile.PL
	make
	make test 
	make install

SUPPORT AND DOCUMENTATION

After installing, you can find documentation for this module with the
perldoc command.

    perldoc Parse::Dia::SQL

You can also look for information at:

    RT, CPAN's request tracker
        http://rt.cpan.org/NoAuth/Bugs.html?Dist=Parse-Dia-SQL

    AnnoCPAN, Annotated CPAN documentation
        http://annocpan.org/dist/Parse-Dia-SQL

    CPAN Ratings
        http://cpanratings.perl.org/d/Parse-Dia-SQL

    Search CPAN
        http://search.cpan.org/dist/Parse-Dia-SQL


COPYRIGHT AND LICENCE

Copyright (C) 2008 Andreas Faafeng

This program is released under the following license: gpl



Changes
Revision history for Perl module Parse::Dia::SQL.

0.15 Thu Apr 15 22:32:41 CEST 2010
    - RT #56357 - Detect Dia database shapes and added 
      experimental parsing (excluding associations) 
      (reported by John-Paul Drawneek).
    - Added LICENSE to conform with debian standards.

0.14 Fri Feb  5 20:13:09 CET 2010
    - RT #53782 - Add typemap support (reported by Jani Päijänen).

0.13 Sun Dec 20 13:01:41 CET 2009
    - RT #52755 - Add support for Postgres unique constraints
      using the smallpackage stereotype (reported by shemgp).
    - RT #50906 - Added PostgreSQL synonyms int2, int4, int8 to
      integer, smallint and bigint respectively (reported by Edy
      Incoletti.
    - Added ignore_type_mismatch option which allows foreign keys
      to have a different + type than the primary key it
      references

0.12 Thu Nov 12 10:51:28 CET 2009 
    - RT #51433 - Multipicity (sic) not handled correctly 
      (reported by Edy Incoletti and Jeff Chimene)
    - Ensure all indices are created before foreign key 
      constraints (all db platforms, reported by Jeff Chimene)

0.11 Sun Jul 12 21:25:38 CEST 2009 
    - Bug #47593 Default type for PSQL 'serial' should be 
      'integer' not 'int' (thanks to Mart van de Wege).
    - Added support for Dia 0.97 source files.  Added verions
      checks for each XML element containing it.
      First reported in bug #47040: Associations are 
      not generated in Dia 0.97

0.10 Sat May 16 21:03:57 CEST 2009
    - Use recent (6.50) EU::MM in build to get LICENCE
      MIN_PERL_VERSION, META_MERGE correctly set.

0.09 Sat May 16 15:29:28 CEST 2009
    - SQLite3: Fixed a bug in the update constraint trigger
    - SQLite3: Added 'on delete cascade' support
    - Added support for index options, which are modelled as
      a comment on the operation (index).
    - Fixed bug: Missing associations should bot prevent 
      index creation.

0.08 Tue Apr 14 14:12:00 CEST 2009
    - Bugfix in Utils.pm (sqlite -> sqlite3).
    - Updated Makefile.PL with META_MERGE and LICENSE.
    - Added MANIFEST.SKIP
    - Removed TODO in boilerplate test.

0.07 Wed Apr  1 09:47:50 CEST 2009
    - Added experimental support for SQLite v.3 (thanks to Peter
      Langton).

0.06 Mon Mar 30 09:59:53 CEST 2009
    - Fixed bug 'missing role name on PK end' (thanks to Peter
      Langton).
    - Fixed foreign key bug for Sybase (case sensitivity issue).
    - Removed 'experimental' tag on MySQL MyISAM and InnoDB.
    - Use Text::Table for comment formatting.

0.05  Mon Mar 16 08:58:33 CET 2009
    - Restrict length of foreign keys in drop statements (see 0.04)

0.04  Fri Mar 13 13:13:13 CET 2009
    - Made foreign keys 18 chars or less on DB2 (overridable)
      regardless of name length in dia file.
        (sub _create_constraint_name)
    - InnoDB: Added drop foreign keys support.
    - Reduced number of columns in log output to screen.

0.03  Mon Mar  2 09:57:23 CET 2009
    - Added table_postfix_options (e.g. 'ENGINE=InnoDB')
    - Made documentation more consistent.
    - Removed private subroutines from main documentation.
    - Made sure parsing is finished before outputting,
        and updated tests accordingly.

0.02  Fri Feb 27 10:29:36 CET 2009
    - Fixed package name error.
    - Added generate_small_package_sql() to Output.pm
    - Fixed comma-comment bug.
    - Added CLI client.

0.01  Mon Feb 23 08:18:14 CET 2009
    - original version



Makefile.PL

# $Id: Makefile.PL,v 1.4 2009/04/14 06:32:24 aff Exp $

use strict;
use warnings;

require 5.006;

use ExtUtils::MakeMaker;

my %args = (
  NAME          => 'Parse::Dia::SQL',
  AUTHOR        => 'Andreas Faafeng <aff@cpan.org>',
  VERSION_FROM  => 'lib/Parse/Dia/SQL.pm',
  ABSTRACT_FROM => 'lib/Parse/Dia/SQL.pm',
  EXE_FILES     => ['bin/parsediasql'],
  PL_FILES      => {},
  LICENSE       => 'gpl',
  META_MERGE    => {
    resources =>
      { repository => 'http://tedia2sql.tigris.org/source/browse/tedia2sql/', },
    keywords => [
      qw [Parse Dia SQL DDL RDBMS DB2 Informix Ingres InnoDB MyISAM MySQL Oracle Postgres SQLite3 Sas Sybase]
    ],
  },

  PREREQ_PM => {
    'Data::Dumper'              => 0,
    'Digest::MD5'               => 0,
    'Fatal'                     => 0,
    'File::Spec::Functions'     => 0,
    'File::Temp'                => 0,
    'File::Find'                => 0,
    'Getopt::Long'              => 0,
    'IO::Uncompress::Gunzip'    => 0,
    'Log::Dispatch::FileRotate' => 0,
    'Log::Log4perl'             => 0,
    'POSIX'                     => 0,
    'Test::Exception'           => 0,
    'Test::More'                => 0,
    'Text::Table'               => 0,
    'XML::DOM'                  => 0,
  },
  dist  => { COMPRESS => 'gzip -9f', SUFFIX => 'gz', },
  clean => { FILES    => 'Parse-Dia-SQL-*' },
);

# Use args according to installed version of ExtUtils::MM
my %filter = (
    MIN_PERL_VERSION     => '6.48',
    LICENSE              => '6.48',
    META_MERGE           => '6.46',
    AUTHOR               => '6.07',
    ABSTRACT_FROM        => '6.07',
);


delete $args {$_} for grep {defined $filter {$_} &&
                            $ExtUtils::MakeMaker::VERSION lt $filter {$_}}
                            keys %args;


WriteMakefile %args;

__END__
